### PR TITLE
Sqlalchemy 2.0 syntax

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR moves more (mostly internal) handlers to the Sqlalchemy 2.0 syntax.

### CODE REVIEW

File: skyportal/handlers/api/internal/profile.py [hunk 1]
```diff
@@ -148,96 +148,103 @@ def patch(self, user_id=None):
...
+            ).first()
+            if user is None:
+                return self.error(f'Cannot find User with ID: {user_id}')
 
-        if data.get("first_name") is not None:
-            user.first_name = data.pop("first_name")
+            if data.get("username") is not None:
+                username = data.pop("username").strip()
+                if username == "":
+                    return self.error("Invalid username.")
```
Comment (guynir42) [inline skyportal/handlers/api/internal/profile.py:166]: We don't actually need to check empty string if checking length.

File: skyportal/handlers/api/internal/recent_sources.py [hunk 2]
```diff
@@ -15,83 +15,82 @@
...
+                                "public_url": t.public_url,
+                            }
+                            for t in sorted(s.thumbnails, key=lambda t: t_index(t.type))
+                        ],
+                        'classifications': s.classifications,
+                        'recency_index': recency_index,
+                    }
+                )
 
-        self.verify_and_commit()
```
Comment (guynir42) [inline skyportal/handlers/api/internal/recent_sources.py:95]: Are you sure this works without committing?

File: skyportal/handlers/api/internal/source_views.py [hunk 3]
```diff
@@ -32,52 +32,58 @@ def get_top_source_views_and_ids(self, current_user):
             .filter(SourceView.created_at >= cutoff_day)
             .order_by(desc('views'))
             .limit(max_num_sources)
-        )
+        ).all()
 
-        return q.all()
+        print(results)
```
Comment (guynir42) [inline skyportal/handlers/api/internal/source_views.py:37]:
```suggestion
```

**Review by guynir42 — APPROVED** [review]
Generally looks good. I have a few minor comments:

File: skyportal/handlers/api/internal/source_views.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/handlers/api/internal/source_views.py]: Yup.

File: skyportal/handlers/api/internal/recent_sources.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/internal/recent_sources.py:95]: It's just a get.

File: skyportal/handlers/api/internal/profile.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/internal/profile.py]: good point.

File: skyportal/handlers/api/comment.py [hunk 4]
```diff
@@ -445,7 +448,7 @@ def post(self, associated_resource_type, resource_id):
             elif associated_resource_type.lower() == "earthquake":
                 earthquake_id = resource_id
                 earthquake = session.scalars(
-                    EarthquakeEvent.select(self.current_user).where(
+                    EarthquakeEvent.select(session.user_or_token).where(
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/comment.py:451]: My determination for all our code to look the same ;)

**Pull request merged**